pennylane/templates/subroutines/uccsd_unitary.py

```diff
@@ -0,0 +1,231 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``UCCSDUnitary`` template.
+"""
+import numpy as np
+
+import pennylane as qml
+
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane.templates.decorator import template
+from pennylane.templates.subroutines import DoubleExcitationUnitary, SingleExcitationUnitary
+from pennylane.templates.utils import (
+    check_no_variable,
+    check_shape,
+    check_type,
+    check_wires,
+    get_shape,
+)
+
+
+@template
+def UCCSDUnitary(weights, wires, ph=None, pphh=None, init_state=None):
+    r"""Implements the the Unitary Coupled-Cluster Singles and Doubles (UCCSD) ansatz
+    by applying the
+    :func:`SingleExcitationUnitary <pennylane.templates.subroutine.SingleExcitationUnitary>`
+    and :func:`DoubleExcitationUnitary <pennylane.templates.subroutine.DoubleExcitationUnitary>`
+    templates to exponentiate the coupled-cluster excitation operator. UCCSD is a VQE ansatz
+    commonly used to run quantum chemistry simulations.
```

Comment:
Can we do this, so that in sphinx autosummary tables only the first sentence appears?

```suggestion
    r"""Implements the the Unitary Coupled-Cluster Singles and Doubles (UCCSD) ansatz. 
    
    The UCCSD ansatz calls the
    :func:`SingleExcitationUnitary <pennylane.templates.subroutine.SingleExcitationUnitary>`
    and :func:`DoubleExcitationUnitary <pennylane.templates.subroutine.DoubleExcitationUnitary>`
    templates to exponentiate the coupled-cluster excitation operator. UCCSD is a VQE ansatz
    commonly used to run quantum chemistry simulations.
```

Reply:
nice @mariaschuld, thanks!